addons/survey/models/survey_survey.py

```diff
@@ -181,6 +191,29 @@ def _read_group_states(self, values, domain, order):
         selection = self.env['survey.survey'].fields_get(allfields=['state'])['state']['selection']
         return [s[0] for s in selection]
 
+    @api.onchange('users_login_required', 'certificate')
+    def _onchange_set_certification_give_badge(self):
+        if not self.users_login_required or not self.certificate:
+            self.certification_give_badge = False
+
+    # CRUD
+    @api.model
+    def create(self, vals):
+        if vals.get('certification_give_badge'):
+            if not vals.get('certification_badge_id'):
+                raise UserError('Certification badge must be configured if Give Badge is set.')
+            survey = super(Survey, self).create(vals)
+            survey.sudo()._create_certification_badge_trigger()
+            return survey
+        return super(Survey, self).create(vals)
+
+    def write(self, vals):
```

Comment:
Don't forget to add tests otherwise @awa-odoo will be very very very sad.

Reply:
<3